src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/LambdaBodyLengthCheck.java

```diff
@@ -0,0 +1,234 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2020 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.checks.sizes;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * <p>
+ * Checks lambda body length.
+ * </p>
+ * <p>
+ * Rationale: Similar to anonymous inner classes, if lambda body becomes very long
+ * it is hard to understand and to see the flow of the method
+ * where the lambda is defined. Therefore long lambda body
+ * should usually be extracted to method.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code max} - Specify the maximum number of lines allowed.
+ * Type is {@code int}.
+ * Default value is {@code 10}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check to accept lambda bodies with up to 10 lines:
+ * </p>
+ * <pre>
+ * &lt;module name="LambdaBodyLength"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * class Test {
+ *   Runnable r = () -&gt; { // violation, 11 lines
+ *       System.out.println(2); // line 2 of lambda
+ *       System.out.println(3);
+ *       System.out.println(4);
+ *       System.out.println(5);
+ *       System.out.println(6);
+ *       System.out.println(7);
+ *       System.out.println(8);
+ *       System.out.println(9);
+ *       System.out.println(10);
+ *   }; // line 11
+ *
+ *   Runnable r2 = () -&gt; // violation, 11 lines
+ *       "someString".trim() // line 1 of lambda
+ *                   .trim()
```

Comment:
please use `.concat("2")..concat("3")` , same idea as in previous example, I like it , it is very visual.

Reply:
done